src/docdb/tree/DocDBDatabaseTreeItemBase.ts

```diff
@@ -35,27 +35,27 @@ export abstract class DocDBDatabaseTreeItemBase extends DocDBTreeItemBase<Collec
     }
 
     public get id(): string {
-        return this._database.id;
+        return nonNullProp(this._database, 'id');
     }
 
     public get label(): string {
-        return this._database.id;
+        return nonNullProp(this._database, 'id');
     }
 
     public get link(): string {
-        return this._database._self;
+        return nonNullProp(this._database, '_self');
     }
 
     public get connectionString(): string {
-        return this.parent.connectionString.concat(`;Database=${this._database.id}`);
+        return this.parent.connectionString.concat(`;Database=${this.id}`);
     }
 
     public get databaseName(): string {
         return this._database.id;
     }
 
-    public async getIterator(client: DocumentClient, feedOptions: FeedOptions): Promise<QueryIterator<CollectionMeta>> {
-        return client.readCollections(this.link, feedOptions);
+    public async getIterator(client: CosmosClient, feedOptions: FeedOptions): Promise<QueryIterator<ContainerDefinition & Resource>> {
+        return client.database(this._database.id).containers.readAll(feedOptions);
```

Comment:
So these "readAll" methods don't _actually_ try to list _all_ items right? They still paginate if there's a bunch?

Reply:
It does paginate based on the FeedOptions, and QueryIterator allows for accessing all in response or in batches.